File: 백준/Silver/1002. 터렛/터렛.cs
int n = int.Parse(Console.ReadLine());

        for (int i = 0; i < n; i++)
        {
            double[] r = Array.ConvertAll(Console.ReadLine().Split(), double.Parse);

            double d = Math.Sqrt(Math.Pow(r[3] - r[0], 2) + Math.Pow(r[4] - r[1], 2));

            if (d == 0)
            {
                if (r[2] == r[5])
                    Console.WriteLine(-1);
                else
                    Console.WriteLine(0);
            }
            else
            {
                if (r[2] + r[5] == d || Math.Abs(r[2] - r[5]) == d)
                    Console.WriteLine(1);
                else if (Math.Abs(r[2] - r[5]) < d && d < r[2] + r[5])
                    Console.WriteLine(2);
                else
                    Console.WriteLine(0);
            }
        }

File: 백준/Silver/1002. 터렛/터렛.cc
#include <iostream>
#include <cmath>

using namespace std;

int main()
{
	ios::sync_with_stdio(false);
	cin.tie(nullptr);
	cout.tie(nullptr);

	int t;
	cin >> t;

	while (t--)
	{
		int x1, y1, r1, x2, y2, r2;
		cin >> x1 >> y1 >> r1 >> x2 >> y2 >> r2;

		const double distance = sqrt(pow(x2 - x1, 2) + pow(y2 - y1, 2));

		if (distance == 0)
			cout << (r1 == r2 ? -1 : 0) << '\n';
		else
		{
			if (r1 + r2 == distance || abs(r1 - r2) == distance)
				cout << 1 << '\n';
			else if (abs(r1 - r2) < distance && distance < r1 + r2)
				cout << 2 << '\n';
			else
				cout << 0 << '\n';
		}
	}
}